### ISSUE TITLE
Support case-insensitive collation utf8mb4_general_ci

### ISSUE THREAD

**Issue by knutwannheden** [dolthub/dolt#2161]
It would be nice with collation support in Dolt. Full support for all collations may be a lot of work, but supporting the case-insensitive collation `utf8mb4_general_ci` (or `utf8mb4_unicode_ci` depending on what matches Go better) only is maybe feasible. This would allow for case-insensitive string comparisons in SQL predicates.

**Comment by Hydrocharged** [comment]
Greetings @knutwannheden! We've finally implemented the framework to easily add collations! Although we don't support every collation and character set right now, it is now relatively easy to add them. Along with the framework, I did add support for a few collations, and included `utf8mb4_general_ci` and `utf8mb4_unicode_ci` per your suggestion :D 

https://github.com/dolthub/go-mysql-server/pull/1147

Look for our next Dolt release coming soon!
